app/src/androidTest/kotlin/com/google/samples/apps/nowinandroid/ui/UiTestExtentions.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.samples.apps.nowinandroid.ui
+
+import androidx.annotation.StringRes
+import androidx.compose.ui.test.junit4.AndroidComposeTestRule
+import kotlin.properties.ReadOnlyProperty
+
+fun AndroidComposeTestRule<*, *>.stringResource(@StringRes resId: Int) =
+    ReadOnlyProperty<Any, String> { _, _ -> activity.getString(resId) }
```

Comment:
nit: IMO we should try to keep explicit return types for all public functions

```suggestion
fun AndroidComposeTestRule<*, *>.stringResource(
    @StringRes resId: Int,
): ReadOnlyProperty<Any, String> = ReadOnlyProperty { _, _ -> activity.getString(resId) }
```

Reply:
Fixed